x-pack/plugins/infra/server/services/log_entries/queries/log_entry.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import type { RequestParams } from '@elastic/elasticsearch';
+import * as rt from 'io-ts';
+import { jsonArrayRT } from '../../../../common/typed_json';
+import {
+  commonHitFieldsRT,
+  commonSearchSuccessResponseFieldsRT,
+} from '../../../utils/elasticsearch_runtime_types';
+
+export const createGetLogEntryQuery = (
+  logEntryIndex: string,
+  logEntryId: string,
+  timestampField: string,
+  tiebreakerField: string
+): RequestParams.Search<Record<string, any>> => ({
+  index: logEntryIndex,
+  terminate_after: 1,
+  track_scores: false,
+  track_total_hits: false,
+  body: {
+    size: 1,
+    query: {
+      ids: {
+        values: [logEntryId],
+      },
+    },
+    fields: ['*'],
+    sort: [{ [timestampField]: 'desc' }, { [tiebreakerField]: 'desc' }],
+    _source: false,
+  },
+});
+
+export const logEntryHitRT = rt.intersection([
+  commonHitFieldsRT,
+  rt.type({
+    fields: rt.record(rt.string, jsonArrayRT),
+    sort: rt.tuple([rt.number, rt.number]),
+  }),
+]);
+
+export type LogEntryHit = rt.TypeOf<typeof logEntryHitRT>;
+
+export const getLogEntryResponseRT = rt.intersection([
```

Comment:
🤔 Just a thought:

Somewhere else in the code we are dynamically generating the request types via a function called `createAsyncRequestRTs()`. Would it make sense to do something similar to this response? 

```
export const getLogEntryResponse = createAsyncResponseRT(logEntryHitRT);
```

I would expect the more search strategies we have, the more we will repeat the pattern of this `rt` type.

Reply:
I considered that too, but didn't follow through. While the search strategy response type is relatively static with only the params being generic, the ES query dsl is a very different beast. Some queries will have `aggs`, some will have other parameters and the responses will also vary accordingly. So I struggle to anticipate the usages right now. Maybe we should consider this once we have multiple queries in which manually assembling the `rt.intersection` like this seems like too much duplication?

Reply:
> Maybe we should consider this once we have multiple queries

Makes sense 👍